### PR TITLE
Create MongoDb new Driver using MondoDb/Driver with mongo-php-library and rename MongoDb legacy driver to Mongo Driver.

### PR DESCRIPTION
**Pull request by fabiocmazzo**
Hi,
  I needed to implement a driver for the new MongoDb PHP Driver and would like to share back this improvement. I kept the legacy driver and renamed to Mongo, the new was as Mongodb.

I hope that helps 1% of what you helped me sharing PhpFastCache.

Thank you

### CODE REVIEW

**Comment by Geolim4** [comment]
Hello!

Thanks you I will have a look tonight!

Cheers
Georges.

Le 20 juil. 2016 4:24 AM, "Fabio Covolo Mazzo" <notifications@github.com> a
écrit :

File: src/phpFastCache/Drivers/Mongo/Item.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * @author Georges.L (Geolim4)  <contact@geolim4.com>
+ *
+ */
+
+namespace phpFastCache\Drivers\Mongo;
+
+use phpFastCache\Cache\ExtendedCacheItemInterface;
+use phpFastCache\Cache\ExtendedCacheItemPoolInterface;
+use phpFastCache\Cache\ItemBaseTrait;
+use phpFastCache\Drivers\Mongodb\Driver as MongodbDriver;
```
Comment (Geolim4) [inline src/phpFastCache/Drivers/Mongo/Item.php]: You need to update the USE here, wrong item name used :)

File: src/phpFastCache/Drivers/Mongo/Item.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+{
+    use ItemBaseTrait;
+
+    /**
+     * Item constructor.
+     * @param \phpFastCache\Drivers\Mongodb\Driver $driver
+     * @param $key
+     * @throws \InvalidArgumentException
+     */
+    public function __construct(MongodbDriver $driver, $key)
```
Comment (Geolim4) [inline src/phpFastCache/Drivers/Mongo/Item.php]: Here too

File: src/phpFastCache/Drivers/Mongo/Item.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    /**
+     * @param ExtendedCacheItemPoolInterface $driver
+     * @throws \InvalidArgumentException
+     * @return static
+     */
+    public function setDriver(ExtendedCacheItemPoolInterface $driver)
+    {
+        if ($driver instanceof MongodbDriver) {
```
Comment (Geolim4) [inline src/phpFastCache/Drivers/Mongo/Item.php]: And here :)

File: src/phpFastCache/Drivers/Mongodb/Driver.php [hunk 4]
```diff
@@ -59,11 +61,16 @@ public function __construct(array $config = [])
      */
     public function driverCheck()
     {
-        if(class_exists('MongoDB\Driver\Manager')){
-            trigger_error('PhpFastCache currently only support the pecl Mongo extension.<br />
-            The Support for the MongoDB extension will be added coming soon.', E_USER_ERROR);
+        if(!class_exists('MongoDB\Driver\Manager')){
+            trigger_error('This driver is used to support the pecl MongoDb extension with mongo-php-library.<br />
```
Comment (Geolim4) [inline src/phpFastCache/Drivers/Mongodb/Driver.php]: The mongo-php-library does requires a composer require or it's a pecl's extension ?

**Comment by Geolim4** [comment]
I commented on some commits, also can you Pr on the V5 please ?
Thanks you :)

File: src/phpFastCache/Drivers/Mongodb/Driver.php (hunk 4 above)
Comment (fabiocmazzo) [inline src/phpFastCache/Drivers/Mongodb/Driver.php]: Requires a Composer require, new MongoDB driver is a low-level driver, you have only database commands, without a high level classes (Collection, Database, etc).

Comment (Geolim4) [inline src/phpFastCache/Drivers/Mongodb/Driver.php]: If there is a composer dependence please add it :)

**Comment by Geolim4** [comment]
Are you talking about this library ?
https://github.com/mongodb/mongo-php-library

**Comment by Geolim4** [comment]
@fabiocmazzo up !

**Comment by Geolim4** [comment]
Hello @authorname,

I’m closing this pull request _for now_ because of (inactivity / outdated code / …).

You can always reopen it though! :) 
Please (update the pull request / rebase your branch / clarify …).

Regards,
Georges.L

**Comment by Geolim4** [comment]
Hello @fabiocmazzo 

I finally implemented your code in the V6 here:
https://github.com/PHPSocialNetwork/phpfastcache/commit/64bc5e61654fba5feba8443638a0c655ff1892f1

Thanks for your contribution :)

**Pull request closed without merging**